pkgs/desktops/deepin/dde-file-manager/default.nix

```diff
@@ -0,0 +1,235 @@
+{ stdenv, fetchFromGitHub, fetchpatch, pkgconfig, qmake, qttools,
+  qtmultimedia, qtx11extras, gnome3, gsettings-qt, dtkcore, dtkwidget,
+  deepin-movie-reborn, jemalloc, ffmpegthumbnailer, poppler, taglib,
+  libsecret, kcodecs, deepin-gettext-tools, dde-polkit-agent, deepin,
+  qt5integration, gtk2, qtsvg, file, avfs, polkit, polkit-qt,
+  deepin-shortcut-viewer, xdg-user-dirs, dde-qt-dbus-factory,
+  deepin-terminal, gst-plugins-good, mpv, zlib, libX11, gvfs, glib,
+  xorg, udisks2-qt5, dde-daemon, dde-dock, deepin-anything,
+  autoPatchelfHook }:
+
+stdenv.mkDerivation rec {
+  name = "${pname}-${version}";
+  pname = "dde-file-manager";
+  version = "4.8.4";
+
+  src = fetchFromGitHub {
+    owner = "linuxdeepin";
+    repo = pname;
+    rev = version;
+    sha256 = "0yhy36gkk5nyn56sn1v3hjri22nxs9k7raflbszahm574xpr1akb";
+  };
+
+  nativeBuildInputs = [
+    deepin.setupHook
+    qmake                   # build
+    qttools                 # build
+    pkgconfig               # build
+    deepin-gettext-tools    # build
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    avfs                   # recommended
```

Comment:
avfs code looks problematic here https://github.com/linuxdeepin/dde-file-manager/blob/e8bd4a23647eee871ad1caf8c10c465be2dbd6e7/utils/utils.cpp#L242

Reply:
What is the proper way of fixing the hard coded FHS path `/bin/bash` in `QString cmd = "/bin/bash"`?

And for `ps` in `args << "-c" << "ps -ax -o 'cmd'|grep '.avfs$'"`?

Reply:
I'd use a patch with `runtimeShell` for `bash`,  `procps` for `ps`, and `gnugrep` and or substitution.

More hardcoded paths at https://github.com/linuxdeepin/dde-file-manager/blob/9dace08c2eb697a8b10447f61f974fd16be271fb/dde-file-manager-lib/controllers/fileeventprocessor.cpp#L51

Reply:
Done.